oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2AuthorizationServerConfiguration.java

```diff
@@ -47,12 +48,17 @@ public static void applyDefaultSecurity(HttpSecurity http) throws Exception {
 				new OAuth2AuthorizationServerConfigurer<>();
 		RequestMatcher endpointsMatcher = authorizationServerConfigurer
 				.getEndpointsMatcher();
+		RequestMatcher oidcClientRegistrationEndpointMatcher = authorizationServerConfigurer
+				.getOidcClientRegistrationEndpointMatcher();
```

Comment:
Let's remove `getOidcClientRegistrationEndpointMatcher()` and move the authorization check in  `OidcClientRegistrationAuthenticationProvider` (see comments in `OidcClientRegistrationEndpointFilter`)

Reply:
Fixed